### PR TITLE
Add getLatestWeakSubjectivityCheckpointEpoch function

### LINKED ISSUE #2304: Add `getLatestWeakSubjectivityCheckpointEpoch` function
**Is your feature request related to a problem? Please describe.**

We would like to implement various features surrounding the weak subjectivity state.
In order to do so, we need to implement a few core utility functions, possibly just 1 to begin.

**Describe the solution you'd like**

Implement `get_latest_weak_subjectivity_checkpoint_epoch` as described here:
https://notes.ethereum.org/@adiasg/weak-subjectvity-eth2#Distributing-Weak-Subjectivity-Checkpoint-States

Ideally we would implement twice, once for `allForks.BeaconState`, and another for `CachedBeaconState<allForks.BeaconState>`.

The code can likely live in `packages/beacon-state-transition/src/util/weakSubjectivity.ts` and
 `packages/beacon-state-transition/src/fast/util/weakSubjectivity.ts`

### PR DESCRIPTION
**Pull request by 3xtr4t3rr3str14l**
resolves #2304  for both allForks.BeaconState, and CachedBeaconState<allForks.BeaconState>

### CODE REVIEW

File: packages/beacon-state-transition/src/util/weakSubjectivity.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export function getLatestWeakSubjectivityCheckpointEpoch(
+  config: IBeaconConfig,
+  state: allForks.BeaconState,
+  safetyDecay = 0.1
+): Epoch {
+  const valCount = getActiveValidatorIndices(state, getCurrentEpoch(config, state)).length;
+  return getEpochFromValCount(config, state, safetyDecay, valCount);
+}
+
+export function getEpochFromValCount(
```
Comment (wemeetagain) [inline packages/beacon-state-transition/src/util/weakSubjectivity.ts]: Can you make this function look like this:
```ts
function getWeakSubjectivityCheckpointEpoch(
  config: IBeaconConfig,
  finalizedEpoch: Epoch,
  activeValidatorCount: number,
  safetyDecay = 0.1
): Epoch;
```

**Review by wemeetagain — APPROVED** [review]
:100:

**Review by dapplion — APPROVED** [review]
Looks good! @wemeetagain What's the next step now? Where should this function be used? Consider opening a follow up issue

**Comment by wemeetagain** [comment]
> Looks good! @wemeetagain What's the next step now? Where should this function be used?

Depending on the route we go next, we may not need to use this, its just nice to have.

The end goal is to have infrastructure in place to provide syncing by weak subjectivity checkpoint+state "by default" for our beacon node, similar to the experience we currently give for starting from genesis.

Imo we can break the problem down into a few parts: creating the ws state, distributing the ws state, loading the ws state.

As far as creating the state, we use something like https://github.com/adiasg/eth2-ws-provider (which just uses the standard API) or add a custom API endpoint to retrieve the weak subjectivity state.

As far as serving the state, it would be nice to use IPFS or similar to leverage existing ChainSafe Files tooling.

And whatever the distribution, we will need to extend our CLI to do this fetch of the ws state on initial boot.

**Pull request merged**